packages/flutter_tools/bin/fuchsia_tester.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:args/args.dart';
+import 'package:process/process.dart';
+import 'package:test/src/executable.dart'
+    as test; // ignore: implementation_imports
+
+import '../lib/src/base/common.dart';
+import '../lib/src/base/config.dart';
+import '../lib/src/base/context.dart';
+import '../lib/src/base/file_system.dart';
+import '../lib/src/base/io.dart';
+import '../lib/src/base/logger.dart';
+import '../lib/src/base/os.dart';
+import '../lib/src/base/platform.dart';
+import '../lib/src/cache.dart';
+import '../lib/src/dart/package_map.dart';
+import '../lib/src/globals.dart';
+import '../lib/src/usage.dart';
+import '../lib/src/test/flutter_platform.dart' as loader;
+
+// Note: this was largely inspired by lib/src/commands/test.dart.
+
+const String _kOptionPackages = "packages";
+const String _kOptionShell = "shell";
+const String _kOptionTestDirectory = "test-directory";
+const String _kOptionFlutterRoot = "flutter-root";
+const List<String> _kRequiredOptions = const <String>[
+  _kOptionPackages,
+  _kOptionShell,
+  _kOptionTestDirectory,
+  _kOptionFlutterRoot,
+];
+
+Future<Null> main(List<String> args) async {
+  final AppContext executableContext = new AppContext();
+  executableContext.setVariable(Logger, new StdoutLogger());
+  executableContext.runInZone(() {
+    // Initialize the context with some defaults.
+    context.putIfAbsent(Platform, () => const LocalPlatform());
+    context.putIfAbsent(FileSystem, () => const LocalFileSystem());
+    context.putIfAbsent(ProcessManager, () => const LocalProcessManager());
+    context.putIfAbsent(Logger, () => new StdoutLogger());
+    context.putIfAbsent(Cache, () => new Cache());
+    context.putIfAbsent(Config, () => new Config());
+    context.putIfAbsent(OperatingSystemUtils, () => new OperatingSystemUtils());
+    context.putIfAbsent(Usage, () => new Usage());
+    return run(args);
+  });
```

Comment:
Is there a way to share this code?  I think we have this same list in three places now.